android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +825,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY;
+
+        if (StringUtils.isAnyBlank(fullName))
+            return getDefaultInitials(size);
+
+        return buildInitialsStringFrom(fullName, size).toUpperCase();
     }
 
-    private String getDefaultInitials() {
+    private String getDefaultInitials(final int size) {
+        if (1 > size)
+            return StringUtils.EMPTY;
+
         int userId = this.selectedUser.getId();
-        int userIndex = getUserScaleIndex(userId);
-        return "P" + userIndex + " ";
+        String userIndex = String.valueOf(getUserScaleIndex(userId));
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (userIndex.length() < size)
+            stringBuilder.append('P');
+
+        stringBuilder.append(userIndex);
+
+        final int padding = size - stringBuilder.length();
+
+        if(0 < padding)
+            stringBuilder.append(StringUtils.repeat(StringUtils.SPACE, padding));
+
+        stringBuilder.setLength(size);
+
+        return stringBuilder.toString();
     }
 
-    private String buildInitialsStringFrom(String fullName) {
-        String[] name = fullName.trim().split(" +");
-        String initials = "";
-        for (int i = 0; i < 3; i++) {
-            if (i < name.length && name[i] != "") {
-                initials += name[i].charAt(0);
-            } else {
-                initials += " ";
-            }
-        }
-        return initials;
+    private String buildInitialsStringFrom(final String fullName, final int size) {
+        String[] names = StringUtils.split(fullName);
+
+        if (null == names || 0 == names.length)
+            return StringUtils.repeat(StringUtils.SPACE, size);
+
+        StringBuilder initialsBuilder = new StringBuilder();
+
+        for (int i = 0; i < size && i < names.length; i++)
+            if (false == StringUtils.isAnyBlank(names[i]))
+                initialsBuilder.append(names[i].charAt(0));
```

Comment:
In the calling function (`getInitials()` line 832) you already made sure that `fullName` is not null and not white-space only.
StringUtils.split() will generate String array that contains strings without any white-space;
```suggestion
        StringBuilder initialsBuilder = new StringBuilder();
        for (int i = 0; i < size && i < names.length; i++)
            initialsBuilder.append(names[i].charAt(0));
```

Reply:
The first check is for the names. Second check inside for is for each item. 

The first check is for the container and we might have something like `names = null` or `names = new String[]()` and that is what is checked against.

Second check is for all elements and we might have `names[i] = null` or `names[i] = new String()`.

They are there for different null pointer exception prevention and we should keep them both.